src/sinol_make/commands/run/__init__.py

```diff
@@ -360,78 +367,184 @@ def print_view(output_file=None):
+			used_solutions = config_expected_scores.keys()
+
+		used_groups = set()
+		if self.args.tests == None and config_expected_scores: # If no groups were specified, use all groups from config
+			for solution in config_expected_scores.keys():
+				for group in config_expected_scores[solution]["expected"]:
+					used_groups.add(group)
+		else:
+			for solution in results.keys():
+				for group in results[solution].keys():
+					used_groups.add(group)
+		used_groups = list(used_groups)
+
+		expected_scores = {} # Expected scores from config with only solutions and groups that were run
+		for solution in used_solutions:
+			if solution in config_expected_scores.keys():
+				expected_scores[solution] = {
+					"expected": {},
+					"points": 0
+				}
+
+				for group in used_groups:
+					if group in config_expected_scores[solution]["expected"]:
+						expected_scores[solution]["expected"][group] = config_expected_scores[solution]["expected"][group]
+
+				expected_scores[solution]["points"] = self.calculate_points(expected_scores[solution]["expected"])
+
+		print(util.bold("Expected scores from config:"))
+		self.print_expected_scores(expected_scores)
+		print(util.bold("\nExpected scores based on results:"))
+		self.print_expected_scores(new_expected_scores)
+
+		expected_scores_diff = dictdiffer.diff(expected_scores, new_expected_scores)
+		added_solutions = set()
+		removed_solutions = set()
+		added_groups = set()
+		removed_groups = set()
+		changes = []
+
+		for type, field, change in list(expected_scores_diff):
+			if type == "add":
+				if field == '': # Solutions were added
+					for solution in change:
+						added_solutions.add(solution[0])
+				elif field[1] == "expected": # Groups were added
+					for group in change:
+						added_groups.add(group[0])
+			elif type == "remove":
+				# We check whether a solution was removed only when sinol_make was run on all of them
+				if field == '' and self.args.programs == None and config_expected_scores:
+					for solution in change:
+						removed_solutions.add(solution[0])
+				# We check whether a group was removed only when sinol_make was run on all of them
+				elif field[1] == "expected" and self.args.tests == None and config_expected_scores:
+					for group in change:
+						removed_groups.add(group[0])
+			elif type == "change":
+				if field[1] == "expected": # Results for at least one group has changed
+					solution = field[0]
+					group = field[2]
+					old_result = change[0]
+					result = change[1]
+					changes.append(ResultChange(solution, group, result, old_result))
+
+		return ValidationResult(
+			added_solutions,
+			removed_solutions,
+			added_groups,
+			removed_groups,
+			changes,
+			expected_scores,
+			new_expected_scores
+		)
+
+
+	def print_expected_scores_diff(self, validation_results: ValidationResult):
+		diff = validation_results
+		config_expected_scores = self.config["sinol_expected_scores"] if "sinol_expected_scores" in self.config else {}
```

Comment:
I bet there is a Python function that retrieves a field in the dictionary when it exists and uses a default value otherwise. Using such a function here would make this line shorter.

Reply:
I found a nice way to do that f5f6eeb8c8faa875898c36d6d86290691f1ab8f3